drools-wb-screens/drools-wb-pmml-text-editor/drools-wb-pmml-text-editor-client/src/main/resources/META-INF/beans.xml

```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright 2019 Red Hat, Inc. and/or its affiliates.
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~  
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~  
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<beans xmlns="http://xmlns.jcp.org/xml/ns/javaee" bean-discovery-mode="all">
+  <scan>
+    <exclude name="org.drools.workbench.screens.pmmltext.client.**"/>
```

Comment:
What does it mean? Is it an exclusion? How `PMMLResourceType` can work considering that is annotated as `@ApplicationScoped` but the package will be excluded?

Reply:
Well....... (other than a copy and paste of the same type of thing from other `-client` modules) I believe it prevents WELD (on the server) from scanning classes that are purely client-side and handled by Errai CDI.

Reply:
Used mainly in `SuperDevMode` when both server and client classes are on the (IDE) classpath.

Reply:
Ok, this means Errai doesn't consider for scanning J2EE CDI configurations (= beans.xml) but only its own file. Nice to know :)

Reply:
Yes, Errai uses `ErraiApp.properties` see [here](http://docs.jboss.org/errai/latest/errai/reference/html_single/#sid-19398999) for "bean scanning".